src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -400,12 +402,12 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
 
   /** Reset company type values to original. */
   resetType () {
-    this.setEntityType(this.getEntitySnapshot?.businessInfo?.legalType)
+    this.setEntityType(this.getOriginalLegalType || null)
     // reset name request
     this.setNameRequest({
-      legalType: this.getEntitySnapshot?.businessInfo?.legalType,
-      legalName: this.getEntitySnapshot?.businessInfo?.legalName,
-      nrNumber: this.getEntitySnapshot?.businessInfo?.nrNumber
+      legalType: this.getOriginalLegalType,
+      legalName: this.getOriginalLegalName,
+      nrNumber: this.getOriginalNrNumber
```

Comment:
After thinking about it and chatting with @argush3 , I am pretty sure that the in-memory NR object should contain the firm's operating name, so that's what this does. There's similar code in filing-template-mixin.ts where resumed filings (drafts) are parsed.